pippy/hf/_SaveModule.py

```diff
@@ -0,0 +1,61 @@
+import torch.distributed as dist
+import pippy
+
+import json
+import os
+
+CKPT_INDEX_JSON_FILENAME = "pytorch_model.bin.index.json"
+
+
+def _save_index(
+    pipe: pippy.fx.GraphModule,
+    ckpt_index_filename: str = CKPT_INDEX_JSON_FILENAME,
+    checkpoint_dir: str = "checkpoints",
+) -> None:
+    """
+    Saves index file describing location of weights in checkpoint.
+
+    Args:
+        pipe (pippy.fx.GraphModule): pipeline graph module with weights to save
+        ckpt_index_filename (str, optional): name of index file. Defaults to "pytorch_model.bin.index.json".
+        checkpoint_dir (str, optional): directory to save checkpoint to. Defaults to "checkpoints".
+    """
+    index_dict = {}
+    total_size = 0
+    index_dict["metadata"] = {"total_size": total_size}
+
+    weight_map = {}
+    for idx, (submod_name, submod) in enumerate(pipe.split_gm.named_children()):
+        for param_name, _ in submod.named_parameters():
+            old_name = submod.remap_qualname(param_name)
+
+            binary_filename = create_binary_filename(idx)
+            weight_map[old_name] = binary_filename
+    index_dict["weight_map"] = weight_map
+
+    json_str = json.dumps(index_dict, indent=4)
+
+    filepath = os.path.join(checkpoint_dir, ckpt_index_filename)
+
+    # create checkpoint directory if it doesn't exist
+    if not os.path.exists(checkpoint_dir):
+        os.mkdir(checkpoint_dir)
+
+    with open(filepath, "w") as f:
+        f.write(json_str)
+
+
+def create_binary_filename(cur_idx: int) -> str:
```

Comment:
nit: since this is gonna be an internal util file, let's do _create_binary_filename(). Actually, _get_binary_filename() will be better, since we will be re-using this when actually saving the parameters.